pymodbus/transaction/transaction.py

```diff
@@ -152,6 +152,7 @@ async def execute(self, no_response_expected: bool, request: ModbusPDU) -> Modbu
                         )
                     return response
                 except asyncio.exceptions.TimeoutError:
+                    self.recv_buffer = b""
```

Comment:
This is the wrong place, since it can cause conflicts with the active callback.

It would be better placed in line 140, which is before the callback is active.

Reply:
Ah, yeah I guess purging the buffer before doing the write makes more sense.